File: lib/models/location.dart
class Location {
  final double latitude;
  final double longitude;
  final String? address;
  final String? placeId;
  final String? locality;
  final String? city;

  Location({
    required this.latitude,
    required this.longitude,
    this.address,
    this.placeId,
    this.locality,
    this.city,
  });

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      address: json['address'] as String?,
      placeId: json['placeId'] as String?,
      locality: json['locality'] as String?,
      city: json['city'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'latitude': latitude,
      'longitude': longitude,
      'address': address,
      'placeId': placeId,
      'locality': locality,
      'city': city,
    };
  }

  @override
  String toString() {
    return address ?? '$latitude, $longitude';
  }
} 

File: backend/app/models/location.py
from pydantic import BaseModel, Field

class Location(BaseModel):
    latitude: float = Field(..., description="Latitude coordinate")
    longitude: float = Field(..., description="Longitude coordinate")
    address: str | None = Field(None, description="Full address")
    place_id: str | None = Field(None, description="Place ID from mapping service")
    locality: str | None = Field(None, description="Locality or area name")
    city: str | None = Field(None, description="City name") 